types/validator_set.go

```diff
@@ -79,6 +79,24 @@ func NewValidatorSet(valz []*Validator) *ValidatorSet {
 	return vals
 }
 
+func (vals *ValidatorSet) ValidateBasic() error {
+	if vals.IsNilOrEmpty() {
+		return errors.New("validator set is nil or empty")
+	}
+
+	for _, val := range vals.Validators {
+		if err := val.ValidateBasic(); err != nil {
+			return fmt.Errorf("validator failed validate basic in set, error: %w", err)
+		}
+	}
+
+	if err := vals.Proposer.ValidateBasic(); err != nil {
+		return fmt.Errorf("proposer failed validate basic, error: %w", err)
+	}
+
```

Comment:
should probably add totalVotingPower > 0

Reply:
Do you think so - it's supposed to be unexported also we ensure that the voting power of each validator isn't less than zero so it can only be zero if all the validators are zero